compiler/noirc_frontend/src/parser/parser/statement.rs

```diff
@@ -7,26 +7,32 @@ use crate::{
         StatementKind, WhileStatement,
     },
     parser::{ParserErrorReason, labels::ParsingRuleLabel},
-    token::{Attribute, Keyword, Token, TokenKind},
+    token::{Attribute, CfgAttribute, Keyword, SecondaryAttribute, Token, TokenKind},
 };
 
 use super::Parser;
 
 impl Parser<'_> {
-    pub(crate) fn parse_statement_or_error(&mut self) -> Statement {
-        if let Some((statement, (_token, _span))) = self.parse_statement() {
-            statement
+    pub(crate) fn parse_statement_or_error(&mut self) -> (Statement, Option<CfgAttribute>) {
+        if let Some((statement, (_token, _location), cfg_attribute)) = self.parse_statement() {
+            (statement, cfg_attribute)
         } else {
             self.expected_label(ParsingRuleLabel::Statement);
-            Statement {
-                kind: StatementKind::Error,
-                location: self.location_at_previous_token_end(),
-            }
+            (
+                Statement {
+                    kind: StatementKind::Error,
+                    location: self.location_at_previous_token_end(),
+                },
+                None,
+            )
         }
     }
 
     /// Statement = Attributes StatementKind ';'?
-    pub(crate) fn parse_statement(&mut self) -> Option<(Statement, (Option<Token>, Location))> {
+    #[allow(clippy::type_complexity)]
+    pub(crate) fn parse_statement(
+        &mut self,
+    ) -> Option<(Statement, (Option<Token>, Location), Option<CfgAttribute>)> {
```

Comment:
Here too, we could have `Statement` hold a list of `CfgAttribute`. Then we'd also filter those at some point (not sure when, maybe during elaboration).

Reply:
I think this is another place having CfgAttribute wrap an existing item would be cleaner instead of making it a separate tuple element. Doesn't touch existing logic but also forces it to be checked since you can't get the data inside otherwise.